src/02-js/01-ordenar/index.js

```diff
@@ -0,0 +1,58 @@
+// CASO 1
+/*const entrada = [456, 234, 78, 2, 8, 90, -12, 4, -23, 0, -100, 50];
+function ordenarMen(a,b){ return a-b; }
+console.log(entrada.sort(ordenarMen));
+
+function ordenarMay(a,b){ return b-a; }
+console.log(entrada.sort(ordenarMay));*/
```

Comment:
@Camila1953, sugiero el siguiente cambio para el caso 1.
Creo que con los comentarios dejé bastante explicado cosas que tal vez te puedan llamar la atención, cualquier cosa me podes preguntar por acá o como ya estamos cerca del finde podemos hacer pair programming como el finde pasado.
Trata de todo lo que es código escribirlo siempre en Inglés, los comentarios es indistinto, pero el código siempre debería estar en Inglés.
Una frase que siempre deberías recordar: "Menos líneas o código, no siempre es mejor".

```suggestion
// Creo esta función para facilitar la ejecución del script y así ejecutar todos los casos sin tener que estar modificando el código
// Con esta función también estoy creando un scope para que en la ejecución de otros casos pueda usar los mismos nombres como input, output, etc
function caseOne() {
  console.log('============= Case 1 ============='); // Agrego este console.log para facilitar la lectura por consola
  // Creo la función customSort porque recordá que el enunciado de este ejercicio decía lo siguiente:
  // "Para cada caso escribir una sola función lo suficientemente flexible para que sea capaz de devolver el array ordenado de forma ascendente o descendente."
  // Con order="ASC" hago que el argumento order sea opcional y en el caso que no se indique por defecto será "ASC"
  function customSort(input, order="ASC") {
    function ascending(a,b) { return a-b; }; // Recordá que siempre podes crear funciones dentro de funciones, para esto casi no hay límite y además podes manejar mejor los scopes (alcances de variables y funciones)
    const descending = (a,b) => { return b-a; }; // Esto es prácticamente equivalente a lo anterior, solo que es otra forma de hacerlo utilizando funciones anonimas (arrow functions)

    let result;

    // Con [...input] hago una copia superficial de la variable input, ya que la función sort muta la variable original.
    // Si no lo hago, por consola siempre obtendré la última mutación de esta variable.
    // Recorda que siempre podes poner el mouse sobre casi cualquier cosa y VS te intentará dar algo de información.
    // En este caso si pones el mouse sobre la función sort podes leer que hace (sino lo podes averiguar en MDN), ahí dice que muta la variable y devuelve una referencia a la variable original.
    // En la forma que vos lo habías hecho no ocurría porque imprimías el resultado antes de la siguiente mutación.
    // En mi caso ocurre porque primero hago las mutaciones y luego imprimo los resultados, por eso es que es necesario hacer [...input].
    // Spread operator (...): esparce la variable.
    // Corchetes []: simplemente para crear un nuevo array.
    // [...input]: creo un nuevo array con los elementos que contiene el array input.
    if (order === "ASC") {
        result = [...input].sort(ascending);
    } else if (order === "DESC") {
        result = [...input].sort(descending);
    } else {
        result = 'Invalid order. It can be "ASC" or "DESC"';
    }

    return result;
  }
  
  const input = [456, 234, 78, 2, 8, 90, -12, 4, -23, 0, -100, 50];
  
  const outputAsc = customSort(input); // No pongo el segundo parámetro porque es opcional
  const outputDesc = customSort(input, "DESC"); // Acça debo poner el segundo parámetro porque quiero que sea "DESC" y por defecto será "ASC"

  // En los siguientes console.logs no solo imprimo el contenido de una variable sino tambien el nombre o algo para entender que es lo que estoy viendo y saber si es correcto o no el resultado
  console.log('input:', input);
  console.log('outputAsc:', outputAsc);  
  console.log('outputDesc:', outputDesc);
}

caseOne();
```

Reply:
@Camila1953 Porfa responde estos mensajes para saber si lo aplicaste, por el último commit veo que lo aplicaste pero lo pusiste todo comentado, la idea de estos comentarios que agrego es que los respondas, para luego cerrarlos, así sabemos lo que ya revisamos o los puntos que al menos en su momento remarqué para reisar. Otra cosa, la idea de todo el código que se suba es que funcione sin necesidad de cambiar casi nada, en este caso es solo descomentar los casos 1 y 2 pero dejarlos descomentados no entra en conflicto con nada, así que lo mejor sería dejarlos descomentados, además de esta forma al ejecutar el archivo index.js ejecutaría todos los casos del ejercicio lo cual es la intención por eso en la sugerencia de código agregué logs para diferenciar por consola fácilmente donde empieza y termina cada caso